basic sprig integration with possible page URL events

bgila opened the pull request:
i've given Austin, James, and Stephen access. Next up I will suggest some surveys

akrolsmir commented on common/envs/prod.ts:
```diff
@@ -3,6 +3,7 @@ export type EnvConfig = {
   firebaseConfig: FirebaseConfig
   amplitudeApiKey?: string
   twitchBotEndpoint?: string
+  sprigEnvironmentId: string
```
Let's make this optional so other configs (dev, or custom instances) won't need it!

bgila replied:
will do. But wait -- we def want this on custom instances, is there same place i can set the config so that it takes effect there too?

jahooma replied:
Custom instances currently are just on different branches. So you'd have to merge this whole PR into each of them.

I don't think I'd worry about it, but we could potentially get it on the Salem Center instance, I guess!

akrolsmir commented on web/lib/service/analytics.ts:
```diff
@@ -12,6 +12,29 @@ init(ENV_CONFIG.amplitudeApiKey ?? '', undefined, { includeReferrer: true })
+
+try {
+  (function(l,e,a,p) {
+    // @ts-expect-error
```
<img width="1209" alt="image" src="https://user-images.githubusercontent.com/856034/192307926-05f750da-cfc5-4a6f-818c-6dd88882d3cf.png">

Currently getting these errors. Could fix, but I think installing the [Sprig NPM package](https://docs.sprig.com/docs/web-javascript#install-with-npmyarn-beta) might just be better

akrolsmir reviewed (commented):
Thanks for sending this out - excited to try out Sprig!

bgila commented:
ready for another review, if you click the link you will see the sample survey on dev

jahooma commented on web/lib/service/analytics.ts:
```diff
@@ -12,6 +12,37 @@ init(ENV_CONFIG.amplitudeApiKey ?? '', undefined, { includeReferrer: true })
 
 export { track }
 
+// Integrate Sprig
```
I'd put this in a new file under `web/lib/service` called `sprig.ts`. Then, import it here for effects, e.g. `import './sprig'`

jahooma replied:
You can disable typescript for the whole file with `// @ts-nocheck` at the top instead of doing it per line.

jahooma commented on web/lib/service/analytics.ts:
```diff
@@ -33,10 +64,14 @@ export const withTracking =
 export async function identifyUser(userId: string) {
   setUserId(userId)
+  // @ts-expect-error Sprig doesn't yet have a native typescript snippet
+  window.Sprig.setUserId(userId)
```
If you've moved sprig to its own file, you could export the Sprig variable and use it here.
E.g. at the bottom of `/sprig.ts`: `export const Sprig = window.Sprig as any`
Then, this line would be simply: `Sprig.setUserId(userId)`

jahooma reviewed (commented):
Sweet, thanks Barak for the Sprig integration! Just chiming in with a few thoughts.

Saw Austin's comment about using the NPM package and agree that would be better if you want to do that instead of following my comments.

One other note: this PR is adding a `package.lock` but it should be deleted since we don't use npm, we use `yarn` which has a `yarn.lock` file. You can type `yarn` into the command line to install packages.

bgila commented:
> Saw Austin's comment about using the NPM package and agree that would be better if you want to do that instead of following my comments.

The issue with NPM is that it won't automatically update to the latest version. 99% of Sprig customers are on the snippet installation..

Addressing all the comments now

bgila commented:
alright PTAnotherL @jahooma

jahooma reviewed (approved):
Awesome, looks great Barak! Thanks for making those changes.

Looking forward to seeing the results from this integration!

bgila commented:
feel free to merge whenever! the study is launched so that will start sending the study to eligible users (after they visit 4 more times)
